### DISCUSSION TITLE
Start a Root Span with a predefined span context and a span link

### DISCUSSION THREAD

**Discussion by ovidiubuligan** [open-telemetry/opentelemetry-cpp#2367] [Q&A]
I have a situation where I want to start the root span with a custom traceid and a span link .

The problem is that I have not seen an API that allows for me to do this  with a link (since links can only be provided at span start).
Right now I get this ugly semi working linking with  invalid parent span  which should actually be the root span  (also customizing the link type may not be available in the API) .

```cpp
   trace::StartSpanOptions options;
   options.kind = trace::SpanKind::kServer;
   options.parent = predefinedContext;
     auto requestSpan = tracer->StartSpan("VBC_Request"
     , {{linkedSpan->GetContext(), {}}}, options);
```

![image](https://github.com/open-telemetry/opentelemetry-cpp/assets/4757459/97e9c7d5-46ea-4cf9-b18c-cc3b9d86ee6f)

Is there a way that I can actually start the root span with the context an spanLink ?

**Comment by marcalff** [comment]
Regarding:

"The problem is that I have not seen an API that allows for me to do this with a link (since links can only be provided at span start)"

This is in progress, please see:

* https://github.com/open-telemetry/opentelemetry-cpp/pull/2380

**Reply by ovidiubuligan** [reply]
Just for info , in c# there is a `Links` [property ](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.links?view=net-7.0#system-diagnostics-activity-links)that looks like you can append to it  . But from what I tested the links added after `Activity` creation  via the Links property are ignored . Similar'ish problem
